jmh-benchmarks/src/main/java/org/apache/kafka/jmh/util/ConcurrentMapBenchmark.java

```diff
@@ -189,4 +194,60 @@ public void testCopyOnWriteMapEntrySet(Blackhole blackhole) {
             }
         }
     }
+
+    @Benchmark
+    @OperationsPerInvocation(TIMES)
+    public void testPCollectionsImmutableMapGet(Blackhole blackhole) {
+        for (int i = 0; i < TIMES; i++) {
+            if (i % writePerLoops == 0) {
+                // add offset mapSize to ensure computeIfAbsent do add new entry
+                pcollectionsImmutableMap = pcollectionsImmutableMap.updated(i + mapSize, 0);
```

Comment:
That is the issue what I mentioned (in jira). `ImmutableMap` is used to deal with single-write/multi-read scenario. If it is facing the multi-write, this style get-and-set  does not work actually. It is possible to lost update.

Reply:
This method is not thread-safe, we can check the result by add `assertEquals(10000, pcollectionsImmutableMap.size());` to verify.

Reply:
so could you please add single-write/multi-read scenario for those maps? That is a existent use case in kafka

Reply:
> so could you please add single-write/multi-read scenario for those maps? That is a existent use case in kafka

Sure. I am working on this. Please wait a moment.